src/Log/Log.php

```diff
@@ -172,37 +172,30 @@ class Log
     ];
 
     /**
-     * Initializes registry and configurations
+     * Creates registry if doesn't exist and creates all defined logging
+     * adapters if config isn't loaded.
      *
-     * @return void
+     * @return \Cake\Log\LogEngineRegistry
      */
-    protected static function _init(): void
+    protected static function getRegistry(): LogEngineRegistry
```

Comment:
psalm cannot follow init functions and gets very angry that everything is uninitialized. I fixed this by adding a getter in other packages so tried here too.